packages/select/docs/FilterMultiSelect.stories.tsx

```diff
@@ -334,3 +335,92 @@ export const DefaultKaizenSiteDemoWithoutScrollbar = () => {
 }
 
 DefaultKaizenSiteDemoWithoutScrollbar.storyName = "With no scrollbar"
+
+export const Async: ComponentStory<typeof FilterMultiSelect> = args => {
+  const [open, setOpen] = useState(false)
+  const [selectedPeople, setSelectedPeople] = useState<string[]>([])
+  const [searchState, setSearchState] = useState("")
+  const { data, isLoading, fetchNextPage, isFetchingNextPage, hasNextPage } =
+    useInfiniteQuery(
+      ["startrek-sg1", searchState],
+      ({ pageParam = 1 }) =>
+        fetch(
+          `https://swapi.dev/api/people/?page=${pageParam}&search=${searchState}`
+        ).then(res => res.json()) as Promise<{
+          results: Array<{ name: string; url: string }>
+          next: string
+        }>,
+      {
+        enabled: open,
+        keepPreviousData: true,
+        getNextPageParam: lastPage => {
+          if (!lastPage.next) return undefined
+          const url = new URL(lastPage.next)
+          const params = new URLSearchParams(url.searchParams)
+          return params.get("page")
+        },
+      }
+    )
+
+  const people = React.useMemo(
+    () =>
+      data?.pages
+        .flat(1)
+        .map(res => res.results)
+        .flat(1)
+        .map(person => ({ label: person.name, value: person.url })) || [],
+    [data]
+  )
+
+  return (
+    <>
+      <FilterMultiSelect
+        {...args}
+        isLoading={isLoading}
+        loadingSkeleton={FilterMultiSelect.MenuLoadingSkeleton}
+        items={people}
+        trigger={() => (
+          <FilterMultiSelect.TriggerButton
+            selectedOptionLabels={getSelectedOptionLabels(
+              new Set(selectedPeople),
+              people
+            )}
+            label={"People"}
+          />
+        )}
+        onSearchInputChange={searchInput => setSearchState(searchInput)}
+        onOpenChange={isOpen => setOpen(isOpen)}
+        onSelectionChange={keys => {
+          if (keys === "all") {
+            return
```

Comment:
I don't know why keys can be `all`. It sends back all the keys when you select all. It doesn't send this string.